onchain/rollups/test/foundry/consensus/quorum/QuorumFactory.t.sol

```diff
@@ -0,0 +1,161 @@
+        address[] quorumValidators;
+        Quorum quorum;
+    }
+
+    function setUp() public {
+        factory = new QuorumFactory();
+    }
+
+    function testNewQuorum(uint256 _numValidators) public {
+        _numValidators = bound(_numValidators, 2, 50);
+        vm.assume(_numValidators >= 2 && _numValidators <= 50);
+
+        address[] memory quorumValidators = generateValidators(_numValidators);
+        uint256[] memory shares = generateShares(quorumValidators);
+
+        IHistory history = new History(msg.sender);
+
+        vm.recordLogs();
+
+        Quorum quorum = factory.newQuorum(quorumValidators, shares, history);
+
+        emit QuorumCreated(quorumValidators, quorum);
+
+        checkFactoryLogs(quorumValidators, quorum);
+    }
+
+    function testNewQuorumDeterministic(
+        uint256 _numValidators,
+        bytes32 _salt
+    ) public {
+        _numValidators = bound(_numValidators, 2, 50);
+        vm.assume(_numValidators >= 2 && _numValidators <= 50);
+
+        address[] memory quorumValidators = generateValidators(_numValidators);
+        uint256[] memory shares = generateShares(quorumValidators);
+
+        IHistory history = new History(msg.sender);
+
+        address precalculatedAddress = factory.calculateQuorumAddress(
+            quorumValidators,
+            shares,
+            history,
+            _salt
+        );
+
+        vm.recordLogs();
+
+        Quorum quorum = factory.newQuorum(
+            quorumValidators,
+            shares,
+            history,
+            _salt
+        );
+
+        emit QuorumCreated(quorumValidators, quorum);
+
+        // Precalculated address must match actual address
+        assertEq(precalculatedAddress, address(quorum));
+        checkFactoryLogs(quorumValidators, quorum);
+    }
+
+    function testAlreadyDeployedNewQuorumDeterministic(
+        uint256 _numValidators,
+        bytes32 _salt
+    ) public {
+        vm.assume(_numValidators > 1);
+        vm.assume(_numValidators < 50);
+
+        address[] memory quorumValidators = generateValidators(_numValidators);
+        uint256[] memory shares = generateShares(quorumValidators);
+
+        IHistory history = new History(msg.sender);
+
+        factory.newQuorum(quorumValidators, shares, history, _salt);
+
+        //Deploy already deployed quorum
+        vm.expectRevert();
+        factory.newQuorum(quorumValidators, shares, history, _salt);
+    }
+
+    // HELPER FUNCTIONS
+    function generateValidators(
+        uint256 _numValidators
+    ) internal pure returns (address[] memory) {
+        address[] memory validators = new address[](_numValidators);
+        for (uint256 i = 0; i < _numValidators; i++) {
+            validators[i] = vm.addr(i + 1);
+        }
+        return validators;
+    }
+
+    function generateShares(
+        address[] memory validators
+    ) internal pure returns (uint256[] memory) {
+        //generate a random number of shares for each validator
```

Comment:
Please add a space after `//`.
Also, we may remove this comment, as this can be clearly deduced by the code.